tests/framework/extensions/nodes/ec2/ec2.go

```diff
@@ -13,20 +15,28 @@ const (
 	nodeBaseName = "rancher-automation"
 )
 
-// CreateNodes creates `numOfInstances` (and/or `numOfWinInstances` when using multiple node configurations - e.g. Windows nodes) number of ec2 instances
-func CreateNodes(client *rancher.Client, numOfInstances int, numOfWinInstances int, multiconfig bool) (ec2Nodes []*nodes.Node, winEC2Nodes []*nodes.Node, err error) {
+// CreateNodes creates `quantityPerPool[]` number of ec2 instances
+func CreateNodes(client *rancher.Client, quantityPerPool []int32) (ec2Nodes []*nodes.Node, err error) {
 	ec2Client, err := client.GetEC2Client()
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
-
-	for _, config := range ec2Client.ClientConfig.AWSEC2Config {
+	// expecting quantityPerPool to match length of ec2Client.ClientConfig.AWSEC2Config
+	if len(quantityPerPool) != len(ec2Client.ClientConfig.AWSEC2Config) {
+		errorString := fmt.Sprintf(
+			"Number of node roles %d does not match number of ec2 configs %d",
+			len(quantityPerPool), len(ec2Client.ClientConfig.AWSEC2Config))
+		return nil, errors.New(errorString)
+	}
+	// provisioning instances in reverse order to allow windows instances time to become ready
+	for i := len(quantityPerPool) - 1; i >= 0; i-- {
```

Comment:
Does that mean, the windows nodes should always be last in the config?

Reply:
that's correct.  I can make this more explicit in the readme, but typically they are build in order, i.e. 
```
  nodesAndRoles:
  - etcd: true
    quantity: 1
  - controlplane: true
    quantity: 1
  - worker: true
    quantity: 2
  - worker: true
    quantity: 1

```